Jenkinsfile

```diff
@@ -65,14 +65,18 @@ def job = {
     }
 
     stage("Check compilation compatibility with Scala 2.12") {
-        sh "./retry_zinc ./gradlew clean assemble spotlessScalaCheck checkstyleMain checkstyleTest spotbugsMain " +
-                 "--no-daemon --stacktrace -PxmlSpotBugsReport=true -PscalaVersion=2.12"
+        sh """
+            ./retry_zinc ./gradlew clean assemble check -x test \
+                --no-daemon --stacktrace -PxmlSpotBugsReport=true -PscalaVersion=2.12
+           """
     }
 
 
     stage("Compile and validate") {
-        sh "./retry_zinc ./gradlew clean assemble publishToMavenLocal spotlessScalaCheck checkstyleMain checkstyleTest spotbugsMain " +
-                "--no-daemon --stacktrace -PxmlSpotBugsReport=true"
+        sh """
+            ./retry_zinc ./gradlew clean assemble publishToMavenLocal check \
```

Comment:
You need to have `-x test` after `check` or else the tests will run too. Also, you can replace `assemble check -x test` with `build -x test`. Same for the Scala 2.12 bit.